<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SomaDeformerWidget</class>
 <widget class="QWidget" name="SomaDeformerWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>802</width>
    <height>723</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget_Controls">
     <property name="minimumSize">
      <size>
       <width>275</width>
       <height>0</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>196</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="layoutDirection">
      <enum>Qt::LeftToRight</enum>
     </property>
     <property name="tabPosition">
      <enum>QTabWidget::West</enum>
     </property>
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="tab_SomaDeformer">
      <attribute name="title">
       <string>Soma builder</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QPushButton" name="pushButton_FinalizeSoma">
         <property name="text">
          <string>Go to Neurite Generation</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_SphericalSoma">
         <property name="text">
          <string>Use Spherical soma</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_RebuildSoma">
         <property name="text">
          <string>Rebuild (advanced options)</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_MSDeformer_AdvanceParams">
      <attribute name="title">
       <string>Advanced options</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QLabel" name="label_SomaScaleFactor">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;This parameter allows to control the size of the initial soma mesh. Ex a value of 1,1 indicate the soma are 10% bigger than initial soma.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>Soma scale factor (Recomended range 0,8-1,4)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="doubleSpinBox_ScaleFactor">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;This parameter allows to control the size of the initial soma mesh. Ex a value of 1,1 indicate the soma are 10% bigger than initial soma.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="maximum">
          <double>10.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.050000000000000</double>
         </property>
         <property name="value">
          <double>1.000000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_UnCollapseSprings">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;This parameter allow to control the influence of neurites in the hole mesh. Lower values cause that deformations for the dendrites are less sharpy and the initial spherical shape remain less. Moreover, higher values cause more sharpy deformations from dendrites, and maintains the initial shape more&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>Uncollapse springs % (Recomended range 10-40)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spinBox_UnCollapseSprigns">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;This parameter allow to control the influence of neurites in the hole mesh. Lower values cause that deformations for the dendrites are less sharpy and the initial spherical shape remain less. Moreover, higher values cause more sharpy deformations from dendrites, and maintains the initial shape more&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="singleStep">
          <number>1</number>
         </property>
         <property name="value">
          <number>7</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_MS_Simulation_Params">
         <property name="title">
          <string>Simulation Params</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="1" column="0">
           <widget class="QLabel" name="label_Stiffness">
            <property name="text">
             <string>Stiffness</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_MS_Stiffness">
            <property name="maximum">
             <double>1500.000000000000000</double>
            </property>
            <property name="value">
             <double>20.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_MS_Dumping">
            <property name="text">
             <string>Dumping</string>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_MS_Dumping">
            <property name="singleStep">
             <double>0.050000000000000</double>
            </property>
            <property name="value">
             <double>0.200000000000000</double>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_NodeMass">
            <property name="singleStep">
             <double>0.010000000000000</double>
            </property>
            <property name="value">
             <double>0.640000000000000</double>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_MassNode">
            <property name="text">
             <string>System Mass</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_MS_Integrator_dt">
            <property name="text">
             <string>dt</string>
            </property>
           </widget>
          </item>
          <item row="3" column="2">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_MS_Integrator_dt">
            <property name="decimals">
             <number>6</number>
            </property>
            <property name="minimum">
             <double>0.000001000000000</double>
            </property>
            <property name="maximum">
             <double>1.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.001000000000000</double>
            </property>
            <property name="value">
             <double>0.005000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_GenerateFromOption">
         <property name="text">
          <string>Build soma</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_restoreDefaults">
         <property name="text">
          <string>Restore default values</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_NextStep">
         <property name="text">
          <string>Go to Neurite Generation</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_MSDeformer_VeryAdvanceParams">
      <attribute name="title">
       <string>Very advanced params</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QScrollArea" name="scrollArea">
         <property name="widgetResizable">
          <bool>true</bool>
         </property>
         <widget class="QWidget" name="scrollAreaWidgetContents">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>256</width>
            <height>1010</height>
           </rect>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QGroupBox" name="groupBox_ExtraMeshTools">
             <property name="title">
              <string>Extra Mesh Tools</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_3">
              <item>
               <widget class="QGroupBox" name="groupBox_MS_Simulation_Controls">
                <property name="title">
                 <string>Simulation Controls</string>
                </property>
                <layout class="QGridLayout" name="gridLayout_5">
                 <item row="0" column="0">
                  <widget class="QPushButton" name="pushButton_MS_StartSimulation">
                   <property name="text">
                    <string>Start Simulation</string>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="0">
                  <widget class="QPushButton" name="pushButton_MS_StopSimulation">
                   <property name="text">
                    <string>Stop Simulation</string>
                   </property>
                  </widget>
                 </item>
                 <item row="4" column="0">
                  <widget class="QPushButton" name="pushButton_MS_ResetSimulation">
                   <property name="text">
                    <string>Reset Simulation</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="0">
                  <widget class="QPushButton" name="pushButton_MS_SingleStepSimulation">
                   <property name="text">
                    <string>Single Step Simulation</string>
                   </property>
                  </widget>
                 </item>
                 <item row="5" column="0">
                  <widget class="QPushButton" name="pushButton_ReteselDendriticBase">
                   <property name="text">
                    <string>Retesel Dendritic Base</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QCheckBox" name="checkBox_NormalizeModel">
                <property name="text">
                 <string>Normalize Model</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QGroupBox" name="groupBox">
                <property name="title">
                 <string>GroupBox</string>
                </property>
                <layout class="QFormLayout" name="formLayout">
                 <item row="0" column="0" colspan="2">
                  <widget class="QCheckBox" name="checkBox_useSWCFile">
                   <property name="text">
                    <string>Use SWC</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="0" colspan="2">
                  <widget class="QRadioButton" name="radioButton_ModelledSomas">
                   <property name="text">
                    <string>Modelled Soma</string>
                   </property>
                   <property name="checked">
                    <bool>false</bool>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="0" colspan="2">
                  <widget class="QComboBox" name="comboBox_ModelledSomas"/>
                 </item>
                 <item row="3" column="0" colspan="2">
                  <widget class="QRadioButton" name="radioButton_ProceduralSomas">
                   <property name="text">
                    <string>Procedural Soma</string>
                   </property>
                   <property name="checked">
                    <bool>false</bool>
                   </property>
                  </widget>
                 </item>
                 <item row="4" column="0">
                  <widget class="QLabel" name="label_SomaHorResol">
                   <property name="text">
                    <string>Hor. Resol.</string>
                   </property>
                  </widget>
                 </item>
                 <item row="4" column="1">
                  <widget class="QSpinBox" name="spinBox_SomaHorResol">
                   <property name="minimum">
                    <number>3</number>
                   </property>
                   <property name="value">
                    <number>8</number>
                   </property>
                  </widget>
                 </item>
                 <item row="5" column="0">
                  <widget class="QLabel" name="label_SomaVerResol">
                   <property name="text">
                    <string>Ver. Resol.</string>
                   </property>
                  </widget>
                 </item>
                 <item row="5" column="1">
                  <widget class="QSpinBox" name="spinBox_SomaVerResol">
                   <property name="minimum">
                    <number>3</number>
                   </property>
                   <property name="value">
                    <number>8</number>
                   </property>
                  </widget>
                 </item>
                 <item row="6" column="0" colspan="2">
                  <widget class="QRadioButton" name="radioButton_XMLSoma">
                   <property name="text">
                    <string>XMLSoma</string>
                   </property>
                   <property name="checked">
                    <bool>true</bool>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_ImportSWC">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string>Import SWC</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_MS_Integrator_Method">
                <property name="text">
                 <string>Method</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="comboBox_MS_Integrator_Method"/>
              </item>
              <item>
               <widget class="QCheckBox" name="checkBox_MS_Constraints">
                <property name="text">
                 <string>Constraints</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QCheckBox" name="checkBox_ExportOBJSecuence">
                <property name="text">
                 <string>Export OBJ Secuence</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QCheckBox" name="checkBox_UseContourPropagation">
                <property name="text">
                 <string>Contour propagation</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QCheckBox" name="checkBox_UseSphericalProjection">
                <property name="text">
                 <string>Spherical Projection</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QCheckBox" name="checkBox_DeleteRedundantVertex">
                <property name="text">
                 <string>Delete Redundant Vertexs</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QGroupBox" name="groupBox_MS_ExternalForce">
                <property name="title">
                 <string>External Forces</string>
                </property>
                <layout class="QFormLayout" name="formLayout_2">
                 <item row="0" column="0">
                  <widget class="QLabel" name="label_gravity">
                   <property name="text">
                    <string>Gravity</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="0">
                  <widget class="QLabel" name="label_X">
                   <property name="text">
                    <string>        X</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="1">
                  <widget class="QDoubleSpinBox" name="doubleSpinBox_gravityForce_X">
                   <property name="decimals">
                    <number>4</number>
                   </property>
                   <property name="minimum">
                    <double>-100.000000000000000</double>
                   </property>
                   <property name="maximum">
                    <double>100.000000000000000</double>
                   </property>
                   <property name="singleStep">
                    <double>0.010000000000000</double>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="0">
                  <widget class="QLabel" name="label_Y">
                   <property name="text">
                    <string>        Y</string>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="1">
                  <widget class="QDoubleSpinBox" name="doubleSpinBox_gravityForce_Y">
                   <property name="decimals">
                    <number>4</number>
                   </property>
                   <property name="minimum">
                    <double>-100.000000000000000</double>
                   </property>
                   <property name="maximum">
                    <double>100.000000000000000</double>
                   </property>
                   <property name="singleStep">
                    <double>0.010000000000000</double>
                   </property>
                   <property name="value">
                    <double>0.000000000000000</double>
                   </property>
                  </widget>
                 </item>
                 <item row="3" column="0">
                  <widget class="QLabel" name="label_Z">
                   <property name="text">
                    <string>       Z</string>
                   </property>
                  </widget>
                 </item>
                 <item row="3" column="1">
                  <widget class="QDoubleSpinBox" name="doubleSpinBox_gravityForce_Z">
                   <property name="decimals">
                    <number>4</number>
                   </property>
                   <property name="minimum">
                    <double>-100.000000000000000</double>
                   </property>
                   <property name="maximum">
                    <double>100.000000000000000</double>
                   </property>
                   <property name="singleStep">
                    <double>0.010000000000000</double>
                   </property>
                  </widget>
                 </item>
                 <item row="4" column="0">
                  <widget class="QLabel" name="label_ApicalParam">
                   <property name="text">
                    <string>ApicalParam</string>
                   </property>
                  </widget>
                 </item>
                 <item row="4" column="1">
                  <widget class="QDoubleSpinBox" name="doubleSpinBox_ApicalParam"/>
                 </item>
                 <item row="5" column="0">
                  <widget class="QLabel" name="label_basalParam">
                   <property name="text">
                    <string>Basal Param</string>
                   </property>
                  </widget>
                 </item>
                 <item row="5" column="1">
                  <widget class="QDoubleSpinBox" name="doubleSpinBox_BasalParam"/>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushButton_UsePatternMesh">
                <property name="text">
                 <string>Use Base Mesh</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_RenderCtrl">
      <attribute name="title">
       <string>Render options.</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QLabel" name="label_AnimationPreriod">
         <property name="text">
          <string>Time to update (msg)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spinBox_animationPeriod">
         <property name="value">
          <number>20</number>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_RenderArea">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
     <property name="title">
      <string/>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_6">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_Viewer"/>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
